packages/medusa/src/scripts/oas/build-oas.ts

```diff
@@ -0,0 +1,124 @@
+import * as path from "path"
+import * as fs from "fs"
+
+import swaggerInline from "swagger-inline"
+import OpenAPIParser from "@readme/openapi-parser"
+import { defaultMetadataStorage } from "class-transformer/cjs/storage"
+
+import { IOptions } from "class-validator-jsonschema/src/options"
+import { validationMetadatasToSchemas } from "class-validator-jsonschema"
+import { IsTypeJSONSchemaConverter } from "../../utils/validators/is-type"
+import { IsNullableJSONSchemaConverter } from "../../utils/validators/is-nullable"
+
+type ApiType = "store" | "admin"
+
+const cliParams = process.argv.slice(2)
+const skipJSONSchema = cliParams.includes("--skipJSONSchema")
+const isVerbose = cliParams.includes("--verbose") || cliParams.includes("-V")
+const debug = (...args) => {
+  if (isVerbose) {
+    console.debug(...args)
+  }
+}
+
+const run = async () => {
+  debug("Generate OAS from codebase.")
+
+  const targetDir = path.resolve(__dirname, "../../../", "dist/oas")
+  fs.mkdirSync(targetDir, { recursive: true })
+
+  for (const apiType of ["store", "admin"]) {
+    debug(`Building OAS for ${apiType} api.`)
+    const oas = await getOASFromCodebase(apiType as ApiType)
+    exportOASToJSON(oas, apiType as ApiType, targetDir)
+  }
+
+  debug("OAS successfully generated.")
+}
+
+const getOASFromCodebase = async (apiType: ApiType) => {
+  debug("Parse JSDoc from path and schema definitions.")
+  const gen = await swaggerInline(
+    [
+      path.resolve(__dirname, "../../", "models"),
+      path.resolve(__dirname, "../../", "api/middlewares"),
+      path.resolve(__dirname, "../../", `api/routes/${apiType}`),
+    ],
+    {
+      base: path.resolve(__dirname, "./", `${apiType}-spec3-base.yaml`),
+      format: ".json",
+    }
+  )
+
+  debug("Get OAS JSON object from OAS JSON string.")
+  const oas = await OpenAPIParser.parse(JSON.parse(gen))
+
+  if (!skipJSONSchema) {
+    debug("Parse Class with class-validator for schema definitions.")
+    const api_ = require("../../api")
+    const models_ = require("../../models")
```

Comment:
suggestion: `await import` instead of require

Reply:
They are not used but I suppose this is because it is not finished yet is that right?

Reply:
Actually, they wouldn't be used. They are required/imported in order for them to get picked up by class-validator and registered in the class-validator global [MetadataStorage](https://github.com/typestack/class-validator/blob/master/src/metadata/MetadataStorage.ts)

Reply:
In that case, just importing them without storing them in a variable would be enough. wdyt?

Reply:
Done